src/main/java/org/testng/internal/ExitCode.java

```diff
@@ -0,0 +1,93 @@
+package org.testng.internal;
+
+import org.testng.IResultMap;
+import org.testng.ITestContext;
+
+import java.util.BitSet;
+
+/**
+ * |---------------------|---------|--------|-------------|------------------------------------------|
+ * | FailedWithinSuccess | Skipped | Failed | Status Code | Remarks                                  |
+ * |---------------------|---------|--------|-------------|------------------------------------------|
+ * | 0                   | 0       | 0      | 0           | Passed tests                             |
+ * | 0                   | 0       | 1      | 1           | Failed tests                             |
+ * | 0                   | 1       | 0      | 2           | Skipped tests                            |
+ * | 0                   | 1       | 1      | 3           | Skipped/Failed tests                     |
+ * | 1                   | 0       | 0      | 4           | FailedWithinSuccess tests                |
+ * | 1                   | 0       | 1      | 5           | FailedWithinSuccess/Failed tests         |
+ * | 1                   | 1       | 0      | 6           | FailedWithinSuccess/Skipped tests        |
+ * | 1                   | 1       | 1      | 7           | FailedWithinSuccess/Skipped/Failed tests |
+ * |---------------------|---------|--------|-------------|------------------------------------------|
+ */
+public class ExitCode {
+
+    private final BitSet exitCodeBits;
+
+    ExitCode() {
+        this(new BitSet(3));
+    }
+
+    public static boolean hasFailureWithinSuccessPercentage(int returnCode) {
+        return (returnCode >= 4 && returnCode <= 7);
```

Comment:
what about 
```java
return (returnCode & 4) == 4;
```

Reply:
where 4 will be a constant

Reply:
same below with skip = 2 and failure = 1

Reply:
`(returnCode & 4) == 4;` would be true only for `4` no ? But the code has to be true for all values between 4 and 7. So I don't think we can use that. Same goes for skip and failure as well, because we now would say an execution had failures, even if it had failures+skips as well.

Reply:
It is a bit mask, so it will return true when the bit is set. It should match with 4, 5, 6, 7.

Same thing for the others.

Reply:
@juherr - I am sorry. I didn't understand that. Can you please explain a bit more in detail perhaps with an example ?

Reply:
Ok scratch that. Now I understood what you were hinting at.

`7 & 4` would be equal to `111 & 100` and the answer would still be `100`. Got it. Will fix it.